PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,468 @@
+
+                Environment environment;
+                if ("sandbox".equals(configuration.getEnvironment())) {
+                    environment = Environment.SANDBOX;
+                } else if ("production".equals(configuration.getEnvironment())) {
+                    environment = Environment.LIVE;
+                } else {
+                    callback.onResult(new IllegalArgumentException("Invalid PayPal Environment"));
+                    return;
+                }
+
+                // Start PayPalCheckout flow
+                PayPalCheckout.setConfig(
+                    new CheckoutConfig(
+                        activity.getApplication(),
+                        configuration.getPayPalClientId(),
+                        environment,
+                        payPalRequest.getNativeConfig().getReturnUrl()
+                    )
+                );
+
+                registerCallbacks(configuration, payPalRequest, payPalResponse);
+
+                PayPalCheckout.startCheckout(createOrderActions -> {
+                    if (payPalRequest instanceof PayPalCheckoutRequest) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    } else if (payPalRequest instanceof PayPalVaultRequest) {
+                        createOrderActions.setBillingAgreementId(payPalResponse.getPairingId());
+                    } else {
+                        callback.onResult(new Exception("Unsupported action type"));
+                    }
+                });
+                callback.onResult(null);
+            } else {
+                callback.onResult(error);
+            }
+        });
+    }
+
+    private void registerCallbacks(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalCheckout.registerCallbacks(
+                approval -> {
+
+                    PayPalAccount payPalAccount = setupAccount(configuration, payPalRequest, payPalResponse);
+
+                    internalPayPalClient.tokenize(payPalAccount, (payPalAccountNonce, error) -> {
+                        if (payPalAccountNonce != null) {
+                            listener.onPayPalSuccess(payPalAccountNonce);
+                        } else {
+                            listener.onPayPalFailure(new Exception("PaypalAccountNonce is null"));
+                        }
+                    });
+
+                },
+                () -> listener.onPayPalFailure(new Exception("User has canceled")),
+                errorInfo -> listener.onPayPalFailure(new Exception(errorInfo.getError().getMessage()))
+        );
+    }
+
+    private PayPalAccount setupAccount(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalAccount payPalAccount = new PayPalAccount();
```

Comment:
So this is a really basic version of the one that was provided here: https://github.com/braintree/braintree_android/blob/00b21449e3fee7595c29ce2c5e31001db7439eb1/PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java#L314

I could potentially need to add more but not sure where to get the rest of these params cause 
```
        final Configuration configuration,
        final PayPalRequest payPalRequest,
        final PayPalResponse payPalResponse
```
does not contain the same values as `browserSwitchResult.getRequestMetadata();`

Reply:
A lot of the values come from the callback URL in the OG PayPal module. Are we missing some values?

Reply:
I'm actually not sure, there seems to be a few things inside `getRequestMetadata();` that aren't inside the other requests. I don't think its necessary to perform the checkout but won't know unless the nonce value succeeds